### DISCUSSION TITLE
alert-rules directory not found error while running Prometheus instance.  (Keeping common.yml in alerts-rules directory)

### DISCUSSION THREAD

**Discussion by gkalluru77** [learndevopsonline/learndevopsonline#1633] [Q&A]
IP Prometheus : 52.204.246.94

Getting below error,  running prometheus instance after referring alert rules from alert-rules directory. From error message understood there is no directory created alert-rues. but noticed in the class no such directory alos created...

T**_ASK [prometheus : Copy Alert Rules] *******************************************
fatal: [localhost]: FAILED! => {"changed": false, "checksum": "3801b7f9207c5c1763ba24e80744ef68fce61c6c", "msg": "Destination directory /opt/prometheus/alert-rules does not exist"}_**

52.204.246.94 | 172.31.42.111 | t3.small | null
[ root@prometheus ~ ]# ansible-pull -i localhost, -U https://github.com/gkalluru77/expense-ansible.git expense.yml -e role_name=prometheus
Starting Ansible Pull at 2023-11-24 07:24:47
/usr/local/bin/ansible-pull -i localhost, -U https://github.com/gkalluru77/expense-ansible.git expense.yml -e role_name=prometheus
[WARNING]: Could not match supplied host pattern, ignoring:
ip-172-31-42-111.ec2.internal
[WARNING]: Could not match supplied host pattern, ignoring: ip-172-31-42-111
localhost | SUCCESS => {
    "after": "8693011f2a5d7fcd60974c56fb84be8b2eae3916",
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/libexec/platform-python"
    },
    "before": "8693011f2a5d7fcd60974c56fb84be8b2eae3916",
    "changed": false,
    "remote_url_changed": false
}
[WARNING]: Could not match supplied host pattern, ignoring:
ip-172-31-42-111.ec2.internal
[WARNING]: Could not match supplied host pattern, ignoring: ip-172-31-42-111

PLAY [Expense Project setup] ***************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [prometheus : Copy Prometheus Configuration] ******************************
ok: [localhost]

T**_ASK [prometheus : Copy Alert Rules] *******************************************
fatal: [localhost]: FAILED! => {"changed": false, "checksum": "3801b7f9207c5c1763ba24e80744ef68fce61c6c", "msg": "Destination directory /opt/prometheus/alert-rules does not exist"}_**

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

**Comment by sreedharm07** [comment]
Hi @gkalluru77 
i think we should create that folder 
please create that folder manually and try, i think it will work properly 
because we given alerts file is under alert-rules/* in prometheus file

**Reply by gkalluru77** [reply]
where i have to create in which folder?
I don't see in Raghu git hub file structure

**Reply by sreedharm07** [reply]
create alert-rules directory ( mkdir alert-rules ) inside **/opt/prometheus/**

**Reply by gkalluru77** [reply]
yes,yes... got it created it... its is fine now...

**Comment by gkalluru77** [comment]
Thanks Sreedhar,  It is working.

**Comment by gkalluru77** [comment]
Issue resolved. hence closing
